File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_of_num = dict()
        for i,num in enumerate(nums):
            if target - num in index_of_num:
                return [i,index_of_num[target - num]]
            index_of_num[num] = i
        

File: 0001-two-sum/0001-two-sum.js
var twoSum = function(nums, target) {
    const N = nums.length;
    const indexOfSeen = new Map();
    
    for(let i=0; i<N; i++){
        let difference = target - nums[i];

        if(indexOfSeen.has(difference)){
            console.log('sdf')
            return [i,indexOfSeen.get(difference)];
        }
        
        indexOfSeen.set(nums[i],i)
    }
    
    console.log(indexOfSeen);
};